### PR TITLE
Github Issue/pull request templates

### PR DESCRIPTION
**Pull request by toogley**
I'm a bit unhappy about the Link for the documentation in the Pull request template, because its now obvious how to add a description there.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+## Are there points in the code the reviewer needs to double check?
+
+## Why was this PR needed?
+
+## Screenshots (if relevant)
+
+## Does this PR meet the acceptance criteria?
+
+     - [ ] [Documentation created/updated](https://www.neomutt.org/guide/)
```
Comment (guyzmo) [inline .github/PULL_REQUEST_TEMPLATE.md]: maybe telling here to modify `doc/manual.xml.head` could be a good idea? (up until we've got a document that says so in http://neomutt.org)

Comment (toogley) [inline .github/PULL_REQUEST_TEMPLATE.md]: ah, great idea.

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+### Actual behavior
+
+### Steps to reproduce
+
+*Please use markdown syntax, especially for Backtraces and such!*
+
+### program versions
+
+    - operating system
+    - neomutt
```
Comment (guyzmo) [inline .github/ISSUE_TEMPLATE.md]: shouldnt' we tell the user to run and give output of `neomutt -v`, and maybe as well of `neomutt -DS`?

Comment (flatcap) [inline .github/ISSUE_TEMPLATE.md]: We need to be a bit careful.
Most people will have the binary `mutt` not `neomutt`.
Most people will have an older version which doesn't have `-D -S`.

**Comment by toogley** [comment]
Would it make sense to suggest using `uname -a` for the infos about the operating system? I don't know sth better which tells the exact OS version and is universal for every OS.

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (toogley) [inline .github/ISSUE_TEMPLATE.md]: @flatcap but then, shouldn't we anyways encourage to update their version? (or at least test if the old version fixes the issue?)

**Comment by guyzmo** [comment]
I think `lsb_release -a` is more helpful, as there's very little chances that knowing the kernel version will help in debugging neomutt, but knowing the exact distribution and release might.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+## Screenshots (if relevant)
+
+## Does this PR meet the acceptance criteria?
+
+     - [ ] Documentation created/updated (you have to edit [doc/manual.xml.head](https://www.neomutt.org/guide/) for that )
+
+     - [ ] All builds are passing
+
+     - [ ] Added [doxygen code documentation](https://www.neomutt.org/dev/doxygen)
```
Comment (guyzmo) [inline .github/PULL_REQUEST_TEMPLATE.md]: maybe a link for doxygen syntax would be nice to have?

**Comment by toogley** [comment]
@guyzmo regarding the use of lsb_release -a: i don't think that is a good idea, because it doesn't work on nixOS which we are supporting.
```
$ lsb_release -a
LSB Version:	n/a
Distributor ID:	n/a
Description:	(none)
Release:	n/a
Codename:	n/a
```

**Comment by flatcap** [comment]
Remember, the instructions are being read by a human :-)

If `lsb_release -a` doesn't work, please tells us what operating system you're using.

**Comment by guyzmo** [comment]
@toogley ok, it's easy: if `lsb_release -a`  returns `n/a` and `(none)` we know it's nixos ☺

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (toogley) [inline .github/ISSUE_TEMPLATE.md]: That's fixed now.

**Comment by toogley** [comment]
I'd like to have feedback on the updated version.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (toogley) [inline .github/PULL_REQUEST_TEMPLATE.md]: Fixed.

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+Tips for a good bugreport
+=========================
+
+* Remember, if you are pedantically detailed about your problem, we can help you
+  better.
```
Comment (guyzmo) [inline .github/ISSUE_TEMPLATE.md]: I'd rather say something like:

* Help us help you! The more details you'll give, the better we'll be able to help you out.

being *pedantically* detailed might be understood negatively.

File: .github/ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+* Please use markdown syntax, especially for Backtraces and config files!
+* Please provide a minimal working example. That means, you should cut out
+  irrelevant code from your muttrc or any other config file relevant to your
+  problem. Especially for complicated issues this is very valuable.
```
Comment (guyzmo) [inline .github/ISSUE_TEMPLATE.md]: I'd rephrase this as:

* Please provide a minimal working example. That means, you should try to replicate your issue with the minimal configuration from your muttrc and/or any other relevant config file. This is the details you should be posting.

the idea of a minimal working example is to help the user think about the problem and try to reduce it to its bare core.

File: .github/ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+* Please give us a list of program versions of **every** program relevant to
+  your report.
+
+Thank you for your help!
+
+*(Of course you can delete this notice after reading it)*
```
Comment (guyzmo) [inline .github/ISSUE_TEMPLATE.md]: s/Of course you can/Please/

**Comment by guyzmo** [comment]
The PR template is 👌 but I got some minor comments to improve the ISSUES template ☺

**Comment by toogley** [comment]
@guyzmo fixed. :)

**Comment by guyzmo** [comment]
BTW, for an enhancement request, I have done some effort on #294 might be a good idea to follow. (if I'm not wrong we can offer multiple templates, can't we?)

**Comment by toogley** [comment]
@guyzmo I don't understand your last sentence. The github templates specifically are about github and have nothing to do with anything mutt does. Therefore i don't understand what both topics have in common.

**Comment by guyzmo** [comment]
I'm talking about the formatting of the ticket:

## Current status

…

## Suggested change

…

## Nice to have

…

## Implementation talk

…

**Comment by toogley** [comment]
@guyzmo Aaah, okay.  But as far as i understood the templates, i don't think there's support for multiple templates to choose from. Do you have a source for your thoughts?^^

**Comment by guyzmo** [comment]
ah… looking at the documentation of github, it's actually a feature of gitlab, my bad ☺

**Comment by toogley** [comment]
@guyzmo hm. I fear it will be too much information, if we add also a "How to write a good Feature Request Issue".  But i consider everything in the current implementation as important - So I don't know how to reduce it.

I'm not sure if it would be a serious alernative, but maybe we can add the information somewhere else? Maybe we should only put two links into the issue template, like so:

```markdown
Here are some Tipps for writing a good...

* bug report [link on neomutt.org]
* feature request [link on neomutt.org]

Please note that our answers will be more satisfying if you respect these tips :)
```

**Comment by toogley** [comment]
@flatcap what do you think about my last answer?

**Comment by flatcap** [comment]
Sorry, for the slow reply.
I haven't merged this because I'm still thinking.
You're right, we don't want to intimidate the poster with a wall of text.

There's an alternative to long `ISSUE_TEMPLATE.md` and `PULL_REQUEST_TEMPLATE.md` files.

If we create `CONTRIBUTING.md`, then when you create an issue, you're presented with a link to the file:
"**Please review the guidelines for contributing to this repository**."

The contributing file could contain detailed instructions on how to fill out the perfect issue/pull-request and the template files could be stripped down.

**Comment by guyzmo** [comment]
I guess a good compromise would be to have a template like:

----------

Please read the guidelines to contributing to this repository: `CONTRIBUTING.md`, and fill up either of the following templates, depending on whether you're reporting a bug, a requesting a feature:

1. for **bug reports**, follow the following template:
  * Expected behavior
  * Actual behavior
  * Steps to reproduce `CONTRIBUTING.md#minimal_example`
  * Used program versions `CONTRIBUTING.md#program_versions`
 
2. for **feature requests** follow the following template:
  * Current status
  * Suggested change
  * Nice to have
  * Implementation talk

------

That way it's short and concise, with links to relevant parts in the CONTRIBUTING document.

**Comment by toogley** [comment]
@guyzmo What do you mean by

* Nice to have
* Implementation talk

**Comment by guyzmo** [comment]
`Nice to have` → extra minor features directly related to the main feature (cf #294)
`Implementation talk` → any first sketchy thought about how to implement the feature
